runtime/commonMain/src/kotlinx/serialization/PlatformUtils.kt

```diff
@@ -7,9 +7,23 @@ package kotlinx.serialization
 import kotlinx.serialization.internal.defaultSerializer
 import kotlin.reflect.KClass
 
+/**
+ * Retrieves a [KSerializer] for the given [KClass].
+ *
+ * Class must be annotated as @Serializable or be one of the built-in types.
+ * Class must have no generic parameters.
+ *
+ * This method may sometimes fail, see [ImplicitReflectionSerializer] documentation.
```

Comment:
While being correct, "may sometimes fail" is not really positive documentation :)
"This method has all the restrictions implied by ImplicitReflectionSerializer"